Objects/abstract.c

```diff
@@ -963,6 +962,46 @@ PyNumber_Add(PyObject *v, PyObject *w)
     return result;
 }
 
+PyObject *
+PyNumber_Divmod(PyObject *v, PyObject *w)
+{
+    PyObject *orig_result = NULL;
+    PyObject *result = binary_op(v, w, NB_SLOT(nb_divmod), "divmod()");
+    if (result == NULL) {
+        return NULL;
+    }
+    if (!PyTuple_Check(result)) {
+        orig_result = result;
+        result = PySequence_Tuple(result);
```

Comment:
You could `Py_DECREF(orig_result)` right here, and then you could eliminate the 3 `Py_(X)DECREF(orig_result)`s below

Reply:
Then there will be an undefined behavior in orig_result != NULL.

Reply:
I missed the connection between `Py_DECREF` and `free` here, good point.

Setting a single boolean in this branch would eliminate the need for the `orig_result != NULL`, and would arguably be clearer - but I suppose that's bike-shedding

Reply:
I now remember why I used this way. Keeping the reference to `orig_result` allows to use it in the error/warning message. The current version does not use it, but I was going to make a separate PR for more detailed error messages for all operations.